pkgs/development/python-modules/pytorch/default.nix

```diff
@@ -156,7 +175,7 @@ in buildPythonPackage rec {
 
   preConfigure = lib.optionalString cudaSupport ''
     export TORCH_CUDA_ARCH_LIST="${lib.strings.concatStringsSep ";" final_cudaArchList}"
-    export CC=${cudatoolkit.cc}/bin/gcc CXX=${cudatoolkit.cc}/bin/g++
+    export CC=${cc}/bin/gcc CXX=${cc}/bin/g++
```

Comment:
What is the motivation for this change? IIUC it's important to use a version of GCC consistent with the CUDA version. How do we handle this in the redist world? Do we have a mechanism? If not, we should look into adding one.

Reply:
EDIT: just saw the code comment above. Yeah, this is a little awkward and keeps us tied to cudatoolkit if we need to always be pulling out `cudaPackages.cudatoolkit.cc`, but I agree that issue is out of scope for this particular PR.

Reply:
Yeah, we need to move the "compatible GCCs` list into the redist expression instead, otherwise it looks kind of weird

Reply:
Actually, I'm really confused why, but `nix-tree` shows that `cudatoolkit` is still part of the closure o_0

Reply:
Oh well... `cudnn.propagatedBuildInputs` 🤦🏻